gemini/prompts/prompt_optimizer/vertex_ai_prompt_optimizer_sdk_custom_metric.ipynb

```diff
@@ -0,0 +1,1390 @@
+        "\n",
+        "    ## Rating rubric\n",
+        "    1 - Minimal: The response lacks personalization and demonstrates minimal engagement with the user. The tone may be impersonal or generic.\n",
+        "    2 - Basic: The response shows some basic personalization but lacks depth or specificity. Engagement is limited, possibly with generic prompts or suggestions. The tone is generally neutral but may lack warmth or empathy.\n",
+        "    3 - Moderate: The response demonstrates clear personalization and attempts to engage the user with relevant follow-up questions or prompts based on their query. The tone is friendly and supportive, fostering a positive user experience.\n",
+        "    4 - High: The response demonstrates a high degree of personalization and actively engages the user with relevant follow-up questions or prompts. The tone is empathetic and understanding, creating a strong connection with the user.\n",
+        "    5 - Exceptional: The response goes above and beyond to personalize the experience, anticipating user needs, and fostering a genuine connection. The tone is warm, encouraging, and inspiring, leaving the user feeling empowered and motivated.\n",
+        "\n",
+        "    ## Evaluation steps\n",
+        "    Step 1: Carefully read both the question and the generated response. Ensure a clear understanding of the user's intent, needs, and any specific context provided.\n",
+        "    Step 2: Evaluate how well the response directly addresses the user's query and demonstrates an understanding of their specific needs or preferences.\n",
+        "    Step 3: Determine the extent to which the response actively encourages further interaction and provides a tailored experience.\n",
+        "    Step 4: Evaluate Tone & Empathy: Analyze the tone of the response, ensuring it fosters a positive and supportive user experience, making the user feel heard and understood.\n",
+        "    Step 5: Based on the three criteria above, assign a score from 1 to 5 according to the score rubric.\n",
+        "    Step 5: Justify the assigned score with a clear and concise explanation, highlighting the strengths and weaknesses of the response with respect to each criterion.\n",
+        "\n",
+        "    # Question : {question}\n",
+        "    # Generated response: {response}\n",
+        "    # Reference response: {target}\n",
+        "    \"\"\"\n",
+        "\n",
+        "    custom_metric_prompt = custom_metric_prompt_template.format(question=question, response=response, target=target)\n",
+        "    response_dict = get_autorater_response(custom_metric_prompt)\n",
+        "\n",
+        "    return {\n",
+        "        \"custom_metric\": response_dict[\"score\"],\n",
+        "        \"explanation\": response_dict[\"explanation\"],\n",
+        "    }\n",
```

Comment:
Is there a built in class to use instead of a generic dict?

Reply:
I couldn't find one.